File: configuration/src/main/java/com/synopsys/integration/configuration/property/types/longs/LongValueParser.java
package com.synopsys.integration.configuration.property.types.longs;

import org.jetbrains.annotations.NotNull;

import com.synopsys.integration.configuration.parse.ValueParseException;
import com.synopsys.integration.configuration.parse.ValueParser;

public class LongValueParser extends ValueParser<Long> {
    @NotNull
    @Override
    public Long parse(@NotNull String value) throws ValueParseException {
        try {
            return Long.parseLong(value);
        } catch (NumberFormatException e) {
            throw new ValueParseException(value, "long", e);
        }
    }
}

File: configuration/src/main/java/com/synopsys/integration/configuration/property/types/longs/LongValueParser.kt
package com.synopsys.integration.configuration.property.types.longs

import com.synopsys.integration.configuration.parse.ValueParseException
import com.synopsys.integration.configuration.parse.ValueParser

class LongValueParser : ValueParser<Long>() {
    override fun parse(value: String) : Long {
        return try {
            value.toLong()
        } catch (e: NumberFormatException) {
            throw ValueParseException(value, "integer", innerException = e)
        }
    }
}